multipaz-models/src/commonMain/kotlin/org/multipaz/models/presentment/mdocPresentment.kt

```diff
@@ -137,6 +149,37 @@ internal suspend fun mdocPresentment(
                     documentTypeRepository = documentTypeRepository,
                     mdocCredential = mdocCredential
                 )
+
+                var zkSystemMatch: ZkSystem? = null
+                var zkSystemSpec: ZkSystemSpec? = null
+                if (usingZkRequest) {
+                    val requesterSupportedZkSpecs = docRequest.getZkSystemSpecs()
+                    if (zkSystemRepository == null) {
+                        val errorMsg = "Reader requested ZK proof but ZK systems are not available."
+                        Logger.e(TAG, errorMsg)
+                        model.setCompleted(Error(errorMsg))
+                        return
+                    }
```

Comment:
In this case we should just fall back to sending a response without a ZK proof, not error out.